packages/types/src/primitive/Extrinsic/SignerPayload.spec.ts

```diff
@@ -2,16 +2,20 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { createType, GenericCall } from '@polkadot/types';
+import { createType } from '@polkadot/types';
 
 import Metadata from '@polkadot/metadata/Metadata';
 import rpcMetadata from '@polkadot/metadata/Metadata/static';
 
 import { TypeRegistry } from '../../codec';
 import SignerPayload from './SignerPayload';
 
+const registry = new TypeRegistry();
+
+// eslint-disable-next-line no-new
+new Metadata(registry, rpcMetadata);
```

Comment:
Could we do `const registry = new TypeRegistry(meta: Metadata | string)` (string being rpc). This way we could remove `setMetadata` from Metadata's constructor, and make it private.

Reply:
hmm, maybe not, because Metadata needs registry already

Reply:
So the above is only in tests - and it is a bit of magic, as per the comment suppression. I was thinking about doing that, but it actually only helps in tests - in the Api itself the registry is created right-up-front before we even have metadata, so we still need to inject or call `setMetadata` (Originally I have an explicit set, now metadata does it).

Your approach may be cleaner, at least for "it not just doing stuff" and will certainly clean up  the tests.

Reply:
Still creates a small issue in the Decorated tests, but can call `setMetadata` there explicitly like we have in the Api then. Ok, will follow the `meta?: string | Uint8Array` and see where it gets us.

Reply:
Yeah, actually I don't think metadata should go inside registry. Ideally, I would see a `class Context` which has this interface:

```
constructor(registry: Registry, metadata?: Metadata);
registry: Registry;
metadata?: Metadata;
setMetadata: (meta: string|8a);
```

In tests that don't need metadata, we just `new Context`. In tests with calls/events, we do an additional `context.setMetadata()`